azure/services/groups/groups.go

```diff
@@ -108,10 +112,15 @@ func (s *Service) IsGroupManaged(ctx context.Context) (bool, error) {
 	defer done()
 
 	groupSpec := s.Scope.GroupSpec()
-	group, err := s.client.Get(ctx, groupSpec.ResourceName())
+	groupIface, err := s.client.Get(ctx, groupSpec)
```

Comment:
you might be able to replace this with `s.Reconciler.Get` or `s.Reconciler.Creator.Get` (I think first one should work) and get rid of the client too